admin/src/main/resources/io/buoyant/admin/js/success_rate_graph.js

```diff
@@ -1,65 +1,76 @@
 /* globals UpdateableChart */
-/* exported ClientSuccessRateGraph */
-var ClientSuccessRateGraph = (function() {
+/* exported SuccessRateGraph */
+var SuccessRateGraph = (function() {
   var columnGridPadding = 27; // from our css grid setup
   var neutralLineColor = "#878787"; // greys.neutral
 
   function createChartLegend(successLineColor) {
     return {
-      referenceLine: neutralLineColor,
       successRate: successLineColor
     }
   }
 
-  function initializeChart($chartEl, dataKeys, timeseriesParamsFn) {
+  function getChartWidthFn(isFullWidthChart, $chartEl) {
+    if (isFullWidthChart) {
+      return function() { return $chartEl.width(); }
+    }
+    else {
+      return function() {
+        var containerWidth = $(".client-container").first().width();
+        var metricsWidth = $(".metrics-container").first().width();
+
+        return containerWidth - metricsWidth - columnGridPadding; // get this to display nicely on wide screens
+      }
+    }
+  }
+
+  function yRangeFunction(range) {
+    var min = range.min === 100 ? 99.99 : range.min;
+    var max = 100.001;
+    return { min: min, max: max };
+  }
+
+  function initializeChart($chartEl, timeseriesParamsFn, chartWidthFn) {
     var $canvas = $("<canvas id='client-success-canvas' height='141'></canvas>");
     $chartEl.append($canvas);
 
     var chart = new UpdateableChart(
       {
-        maxValue: 100.000001, // the 0.000001 is a SmoothieCharts hack
```

Comment:
Yeah, this maxValue doesn't make sense any more now that we're fully customizing the y range.  I got rid of this maxValue - the .001 is so that the graph line markers extend past the 100% success rate line (to make the graph look less confusing in the case of a pure 100% success rate). Before I was using a custom yRangeFunction, I was using the maxValue of 100.000001 to pin the graph's upper limit so that the 100% line would appear at the top and look reasonable, but the custom yRangeFunction does this better. I can pull these out into constants!